src/nodes/hypertable_modify.c

```diff
@@ -745,7 +1018,54 @@ ExecModifyTable(PlanState *pstate)
 	context.mtstate = node;
 	context.epqstate = &node->mt_epqstate;
 	context.estate = estate;
-
+	/*
+	 * For UPDATE/DELETE on compressed hypertable, decompress chunks and
+	 * move rows to uncompressed chunks.
+	 */
+	if (operation == CMD_DELETE || operation == CMD_UPDATE)
+		collect_chunks_from_scan(pstate, ht_state);
+	if ((operation == CMD_DELETE || operation == CMD_UPDATE) && !ht_state->comp_chunks_processed)
+	{
+		ListCell *ch;
+		foreach (ch, ht_state->chunks)
+		{
+			Chunk *chunk = lfirst(ch);
+			bool chunk_status_changed = false;
+			if (ts_cm_functions->decompress_batches_for_update_delete)
```

Comment:
Ok, now i handle by reporting a more general way. Please suggest a better error message.
```
if (result != TM_Ok)
		{
			ereport(ERROR,
					(errcode(ERRCODE_INTERNAL_ERROR),
					 errmsg("Could not perform this operation. Please retry again.")));
			return false;
		}
```

Reply:
I think we can learn from the normal `ExecUpdate`. The correct thing to do is pretty complicated, and depends on the transaction isolation level. Maybe for starters just this simple check is OK. The usual error is:
```
						ereport(ERROR,
								(errcode(ERRCODE_T_R_SERIALIZATION_FAILURE),
								 errmsg("could not serialize access due to concurrent update")));
```
Maybe we could also add an errhint that prints out the tuple values, if it's easy to do. Just to give the users more context.

Let's add at least a very basic test for this, the concurrent modifications are a frequent scenario.

Reply:
I added a testcase in compression_ddl_iso.spec, however since we support this feature only in PG version >= 14
i see testcase fail for PG12 and PG13.

Should i skip this tests for  PG12 and PG13 ?
Or
Please suggest the where can i add concurrent tests ?

Reply:
Let's add a new separate file for compressed update/delete isolation tests, and skip it for older versions.

Reply:
Ok, added new test file compression_dml_iso.spec which will run only for PG versions >= 14